src/test/java/org/sasanlabs/service/vulnerability/commandInjection/CommandInjectionTest.java

```diff
@@ -0,0 +1,125 @@
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.util.Strings;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.sasanlabs.service.exception.ServiceApplicationException;
+import org.sasanlabs.service.vulnerability.bean.GenericVulnerabilityResponseBean;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.RequestEntity;
+import org.springframework.http.ResponseEntity;
+
+/*
+ * @author Feng-Yuan Yang coolesaltw@hotmail.com
+ */
+public class CommandInjectionTest {
+
+    private static CommandInjection commandInjection;
+
+    private static final Supplier<Boolean> trueCondition = () -> true;
+    private static final Supplier<Boolean> falseCondition = () -> false;
+    private static final String successReturnStr = "PING SUCCESS";
+
+    @BeforeEach
+    void setUp() throws IOException {
+        commandInjection = Mockito.spy(new CommandInjection());
+
+        StringBuilder stringBuilder = new StringBuilder().append(successReturnStr);
+
+        // Real localhost ping is not able to operate in CI, Mock instead
+        doReturn(stringBuilder)
+                .when(commandInjection)
+                .getResponseFromPingCommand(eq(LOCALHOST), eq(trueCondition));
+
+        doReturn(trueCondition).when(commandInjection).getSupplier(true);
+        doReturn(falseCondition).when(commandInjection).getSupplier(false);
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_Localhost_ExpectPingSuccess() throws IOException {
+        // Act
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(LOCALHOST);
+
+        // Assert
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(successReturnStr);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(LOCALHOST);
+        verify(commandInjection).getResponseFromPingCommand(eq(LOCALHOST), eq(trueCondition));
+    }
+
+    @Test
+    public void getVulnerablePayloadLevel1_NoIp_ExpectNoResponseContent() throws IOException {
+        ResponseEntity<GenericVulnerabilityResponseBean<String>> responseEntity =
+                commandInjection.getVulnerablePayloadLevel1(Strings.EMPTY);
+
+        // Assert
+        assertThat(responseEntity.getBody().getContent()).isEqualTo(Strings.EMPTY);
+        assertThat(responseEntity.getStatusCode()).isEqualTo(HttpStatus.OK);
+        assertThat(responseEntity.getBody().getIsValid()).isEqualTo(true);
+
+        verify(commandInjection).getVulnerablePayloadLevel1(eq(Strings.EMPTY));
+        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), eq(falseCondition));
```

Comment:
```suggestion
        verify(commandInjection).getResponseFromPingCommand(eq(Strings.EMPTY), (right) -> !right.get());
```

I think this or similar to this should work. Please check once.

Reply:
Please ignore for now.